components/bl-AnimateOnScroll-component/src/index.js

```diff
@@ -0,0 +1,62 @@
+import { useEffect, useMemo, useRef } from 'react';
+import { Aos } from './aos';
+
+const { cn } = BackendlessUI.CSSUtils;
+
+const DevicesMap = {
+  mobile: 'mobile',
+  tablet: 'tablet',
+  all   : true,
+  none  : false,
+};
+
+export default function AnimateOnScroll({ component, eventHandlers, pods }) {
+  const {
+    display, classList, style, animationType, easing, side,
+    duration, offset, disableFor, delay, mirror, once, anchor,
+  } = component;
+  const { onAnimationEnter, onAnimationOut } = eventHandlers;
+  const elementPod = pods['animationPod'];
+
+  const animationOnScrollRef = useRef(null);
+
+  const animationName = useMemo(() => {
+    return animationType + (side === 'none' ? '' : `-${ side }`);
+  }, [animationType, side]);
+
+  useEffect(() => {
+    Aos.init({ disable: DevicesMap[disableFor] });
+
+    document.addEventListener('aos:in', () => onAnimationEnter());
+    document.addEventListener('aos:out', () => onAnimationOut());
+
+    return () => {
+      document.removeEventListener('aos:in', () => onAnimationEnter());
+      document.removeEventListener('aos:out', () => onAnimationOut());
+    }
+  }, []);
+
+  component.el = animationOnScrollRef.current;
```

Comment:
I'm not sure this is the correct way

let's imaging this was rendered only once and the `animationOnScrollRef.current` before first render will be null and as result it won't work 

I assume we need to wrap it with `useEffect`